docs/overrides/main.html

```diff
@@ -3,5 +3,90 @@
+
+        try {
+            posthog.init('phc_WEVBsfcilPS9u6e0CY22d8DqtFt0zalAP9A6iyCBHNH', {
+                api_host: 'https://eu.i.posthog.com',
+                persistence: 'memory'
+            });
+            window.__posthogInitialized = true;
+        } catch (error) {
+            console.error('PostHog initialization failed:', error);
+        }}
+
+document.addEventListener('DOMContentLoaded', function() {
+
+    // Setup
+    const elements = {
+        form: document.querySelector('.md-search__form'),
+        input: document.querySelector('[data-md-component="search-query"]'),
+        results: document.querySelector('.md-search__output')
+    };
+
+    if (!elements.form || !elements.input || !elements.results) return;
+
+    let searchTimeout;
+    let lastSearch = { term: '', time: 0 };
+
+    // Track search with debouncing
+    function trackSearch(term, type) {
+        if (term.length < 3 || term === lastSearch.term || Date.now() - lastSearch.time < 1000) return;
+
+        setTimeout(() => {
+            const resultCount = elements.results.querySelectorAll('.md-search-result__item').length;
+            posthog.capture('documentation_search', {
+                search_term: term,
+                search_type: type,
+                result_count: resultCount,
+                has_results: resultCount > 0,
+                timestamp: Date.now()
+            });
+
+            lastSearch = { term, time: Date.now() };
+        }, 100);
```

Comment:
please extract these numbers to named constants. in particular i can't tell if this 100 is meant to actually be 1000, and another reader might not notice the difference

Reply:
done with explanation